### ISSUE TITLE
Script Generator: 

### ISSUE THREAD

**Issue by adalator** [ISISComputingGroup/IBEX#5425]
Currently, if you don't have a `C:\Scripts` directory then trying to save a generated script fails with an unhelpful error message. There should either be the option to select a directory to save generated scripts to, or the error message should tell the user what the problem is. The former is probably better.

Acceptance Criteria:
- [ ] User can select where to save generated scripts
OR
- [ ] Error message on attempting to save generated script with no `C:\Scripts` directory tells the user to create this directory

**Comment by kjwoodsISIS** [comment]
I agree with @adalator 's comment ("the former is probably better").  I think the best solution would be to use a File Save dialog to allow the user to choose where to save his/her scripts - it is the natural thing to do, most people will be used to such an operation.  On first use, we could default the location to be the `C:\Scripts` folder.

**Comment by ChrisM-S** [comment]
Yes, this makes a lot of sense.  It also make me wonder (w.r.t new Win 10 system) where we should expect `C:\scripts` to be a link to (on Instrument machines only), should this be stored along with configurations etc in `C:\Instrument\Settings\...` where they could be automatically fed into Git, or maybe in `C:\Instrument\var` where they would essentially be expendable (but kept as part of the persistent data "state" of the instrument between updates)?
